### PR TITLE
metrics: split openshift_csi_share_mount_total into total and failures

### PR DESCRIPTION
**Pull request by jan--f**
As proposed in https://github.com/openshift/cluster-monitoring-operator/pull/1477 this spits the metric `openshift_csi_share_mount_total` into `openshift_csi_share_mount_requests_total` and `openshift_csi_share_mount_failures_total`.

Feel free to stick with what is in the code now. I wanted to propose this as I approved the current state in #54.

### CODE REVIEW

**Comment by jan--f** [comment]
cc @alicerum

File: pkg/metrics/metrics.go
Comment (adambkaplan) [inline pkg/metrics/metrics.go:23]: Style nit - pluralize:
```suggestion
func createMountCounters() (prometheus.Counter, prometheus.Counter) {
```

File: pkg/metrics/metrics.go
Comment (adambkaplan) [inline pkg/metrics/metrics.go:20]: Style nit - pluralize
```suggestion
	mountCounter, failedMountCounter = createMountCounters()
```

File: pkg/metrics/metrics.go
Comment (adambkaplan) [inline pkg/metrics/metrics.go:43]: Style nit - we should pluralize this method name to `IncMountCounters` (this is best done in an IDE)

File: pkg/metrics/server_test.go
Comment (adambkaplan) [inline pkg/metrics/server_test.go:163]:
```suggestion
		mountCounter, failedMountCounter = createMountCounters()
```

**Review by adambkaplan — COMMENTED** [review]
/approve

Generally yes to these changes, mainly style nits here. Approving so other team members can LGTM

File: pkg/metrics/metrics.go
Comment (jan--f) [reply pkg/metrics/metrics.go]: Thanks, good catch!

**Comment by alicerum** [comment]
@gabemontero could you please merge this

**Comment by gabemontero** [comment]
looks like we need to employ no feature freeze processes here

/assign @prietyc123 

for qe approve label

/assign @rolfedh 

for docs approve label

/assign @adambkaplan 

I *think* he is able to px approve ... otherwise, hopefully he'll know who to assign these days

**Comment by rolfedh** [comment]
@gabemontero Do these changes have any doc impacts?

**Comment by gabemontero** [comment]
> @gabemontero Do these changes have any doc impacts?

historically we have not described metrics in details in the docs @rolfedh ... we concluded recently that we would maintain that stance with the new shared resource stuff

**Review by rolfedh — APPROVED** [review]

**Comment by gabemontero** [comment]
@alicerum - note that @jan--f 's PR now needs to be rebased (did not get the qe approval label in time to merge)

if he does not have time to rebase, you will need to take his PR here and apply it to the latest master and take over from him.

/label px-approved

**Comment by jan--f** [comment]
Rebased. Fwiw I don't mind if you'll go with @alicerum PR. I just wanted to provide this as my initial review of the new metrics was lacking. I'll leave it up to you folks :)

**Comment by gabemontero** [comment]
On Thu, Dec 2, 2021 at 9:54 AM Jan Fajerski ***@***.***>
wrote:

> Rebased. Fwiw I don't mind if you'll go with @alicerum
> <https://github.com/alicerum> PR. I just wanted to provide this as my
> initial review of the new metrics was lacking. I'll leave it up to you
> folks :)
>
we appreciate it but @alicerum and I talked and decided to go with yours

**Comment by gabemontero** [comment]
with @prietyc123 still not 100% / under the weather 

/assign @jitendar-singh

**Pull request merged**
